_posts/2024-04-29-elytron-oidc-client-scope.adoc

```diff
@@ -0,0 +1,339 @@
+
+. Now, click on *Realm roles* and create two roles, *user* and *admin*. 
+
+. Finally, create a user called *alice* whose *First name* is *Alice* and *Last name* is *Smith* and assign her the *user* and *admin* roles. Steps for assigning roles can be found in the https://www.keycloak.org/docs/latest/server_admin/#proc-assigning-role-mappings_server_administration_guide[ Keycloak documentation]. We will be using different scopes to query these pieces of information about Alice. 
+
+. Next, navigate to the *Credentials* tab and create a password for Alice. Toggle the *Temporary* switch off so you are not prompted to update the password after the first login. 
+
+== Add Helm Configuration
+* Obtain the URL for Keycloak.
+```
+KEYCLOAK_URL=https://$(oc get route keycloak --template='{{ .spec.host }}') &&
+echo "" &&
+echo "Keycloak URL:   $KEYCLOAK_URL" &&
+echo ""
+```
+
+* Switch to the charts directory in the `elytron-oidc-client-scope` example.
+```
+cd /PATH/TO/ELYTRON/EXAMPLES/elytron-oidc-client-scope/charts
+```
+Notice there’s a helm.yaml file in this directory with the following content:
+```
+build:
+  uri: https://github.com/wildfly-security-incubator/elytron-examples.git
+  contextDir: elytron-oidc-client-scope
+deploy:
+  replicas: 1
+  env:
+    - name: OIDC_PROVIDER_URL
+      value: <KEYCLOAK_URL>    <1>
+    - name: SERVER_ARGS
+      value: "--stability=preview" 
+```
+<1> Replace <KEYCLOAK_URL> with the Keycloak URL obtained in the previous command.
+
+== Stability Levels for OpenShift Deployment 
+The WildFly server now includes different stability levels, that can be associated with functionality. Users can use the `--stability` argument when staring the WildFly server. Depending on the value of the stability levels, different features are available. You can learn more about stability levels https://docs.wildfly.org/32/Admin_Guide.html#Feature_stability_levels[here]. 
+
+The `scope` attribute under the `elytron-oidc-client` subsystem is a `preview` level feature, which means in order to access its functionality, the server's stability level must be set to `preview` or lower. When applications are deployed to OpenShift, the WildFly Cloud Galleon Feature Pack is used to provision a server. Therefore, in order to use the scope attribute, we need to provision the server at the `preview` stability level. This is why we have added the environment variable named *SERVER_ARGS* with a value of *--stability=preview*, which specifies that the provisioned server should be started at the `preview` stability level. For more information about the server's stability levels, please refer to https://docs.wildfly.org/32/Admin_Guide.html#Feature_stability_levels[WildFly Docs].
```

Comment:
s/must be set to `preview` or lower/must be set to `preview`

Reply:
Fixed and updated the links to fix 404 errors. 
Thanks @fjuma